doc/sphinx/practical-tools/utilities.rst

```diff
@@ -1,482 +1,981 @@
+    :COQC, COQDEP, COQDOC: 
+      can be set in order to use alternative binaries
+      (e.g. wrappers)
+
+
+:Rule extension: 
   The following makefile rules can be extended. For example
-\begin{verbatim}
-pre-all::
-        echo "This line is print before making the all target"
-install-extra::
-        cp ThisExtraFile /there/it/goes
-\end{verbatim}
-  \begin{description}
-  \item[pre-all::] run before the {\tt all} target.  One can use this
-	  to configure the project, or initialize sub modules or check
-	  dependencies are met.
-  \item[post-all::] run after the {\tt all} target.  One can use this
-	  to run a test suite, or compile extracted code.
-  \item[install-extra::] run after {\tt install}.  One can use this
-	  to install extra files.
-  \item[install-doc::]  One can use this to install extra doc.
-  \item[uninstall::] 
-  \item[uninstall-doc::]
-  \item[clean::]
-  \item[cleanall::]
-  \item[archclean::]
-  \item[merlin-hook::] One can append lines to the generated {\tt .merlin}
-	  file extending this target.
-  \end{description}
-\end{description}
-
-\paragraph{Timing targets and performance testing} %%%%%%%%%%%%%%%%%%%%%%%%%%%
-The generated \texttt{Makefile} supports the generation of two kinds
-of timing data: per-file build-times, and per-line times for an
-individual file.
-
-The following targets and \texttt{Makefile} variables allow collection
-of per-file timing data:
-\begin{itemize}
-\item \texttt{TIMED=1} --- passing this variable will cause
-  \texttt{make} to emit a line describing the user-space build-time
-  and peak memory usage for each file built.
-
-  \texttt{Note}: On Mac OS, this works best if you've installed
-  \texttt{gnu-time}.
-
-  \texttt{Example}: For example, the output of \texttt{make TIMED=1}
-  may look like this:
-\begin{verbatim}
-COQDEP Fast.v
-COQDEP Slow.v
-COQC Slow.v
-Slow (user: 0.34 mem: 395448 ko)
-COQC Fast.v
-Fast (user: 0.01 mem: 45184 ko)
-\end{verbatim}
-\item \texttt{pretty-timed} --- this target stores the output of
-  \texttt{make TIMED=1} into \texttt{time-of-build.log}, and displays
-  a table of the times, sorted from slowest to fastest, which is also
-  stored in \texttt{time-of-build-pretty.log}.  If you want to
-  construct the log for targets other than the default one, you can
-  pass them via the variable \texttt{TGTS}, e.g., \texttt{make
-    pretty-timed TGTS="a.vo b.vo"}.
-
-  \texttt{Note}: This target requires \texttt{python} to build the table.
-
-  \texttt{Note}: This target will \emph{append} to the timing log; if
-  you want a fresh start, you must remove the file
-  \texttt{time-of-build.log} or run \texttt{make cleanall}.
-
-  \texttt{Example}: For example, the output of \texttt{make
-    pretty-timed} may look like this:
-\begin{verbatim}
-COQDEP Fast.v
-COQDEP Slow.v
-COQC Slow.v
-Slow (user: 0.36 mem: 393912 ko)
-COQC Fast.v
-Fast (user: 0.05 mem: 45992 ko)
-Time     | File Name
---------------------
-0m00.41s | Total
+
+  ::
+
+    pre-all::
```

Comment:
These two colons comes from the original text. Unfortunately they seem to be interpreted as starting some verbatim text and should probably be quoted. The same is true for the following targets. Note that I didn't test this but I'm basing my assumption on GitHub's syntax highlighting.